source/agora/consensus/protocol/Nominator.d

```diff
@@ -1520,26 +1520,33 @@ extern(D):
                 auto candidate = cand.getValue();
                 auto data = deserializeFull!ConsensusData(candidate[]);
                 log.trace("Consensus data: {}", data.prettify);
-
-                Amount total_rate;
-                foreach (const ref tx_hash; this.ledger.nominated_tx_sets[data.tx_set])
+                if (auto tx_set = data.tx_set in this.ledger.nominated_tx_sets)
                 {
-                    Amount rate;
-                    auto errormsg = this.ledger.getTxFeeRate(tx_hash, rate);
-                    if (errormsg == NodeLedger.InvalidConsensusDataReason.NotInPool)
-                        continue; // most likely a CoinBase Transaction
-                    else if (errormsg)
-                        assert(0);
-                    total_rate += rate;
-                }
+                    Amount total_rate;
+                    foreach (const ref tx_hash; *tx_set)
+                    {
+                        Amount rate;
+                        auto errormsg = this.ledger.getTxFeeRate(tx_hash, rate);
+                        if (errormsg == NodeLedger.InvalidConsensusDataReason.NotInPool)
+                            continue; // most likely a CoinBase Transaction
+                        else if (errormsg)
+                            assert(0);
+                        total_rate += rate;
+                    }
 
-                CandidateHolder candidate_holder =
+                    CandidateHolder candidate_holder =
+                    {
+                        consensus_data: data,
+                        hash: data.hashFull(),
+                        total_rate: total_rate,
+                    };
+                    candidate_holders ~= candidate_holder;
+                }
+                else
                 {
-                    consensus_data: data,
-                    hash: data.hashFull(),
-                    total_rate = total_rate,
-                };
-                candidate_holders ~= candidate_holder;
+                    log.error("{}: Missing tx set with hash {} in nominated tx sets {}",
```

Comment:
```suggestion
                    log.fatal("{}: Missing tx set with hash {} in nominated tx sets {}",
```
And I think you intend to avoid the assertion so you don't add `assert(0)`, right?

Reply:
This not fatal and was just simply an oversight on my part.

Quorum may reach quorum threshold of votes for a value and consider it for balloting phase before we validate that value and vote on it ourselves.